lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,6 +153,91 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(connection, logger, connection_options, config)
+          super
+          if @connection
+            @verified = true
+            @raw_connection = @connection
+          end
+          # Ensure that we're treating prepared_statements in the same way that Rails 7.1 does
+          @prepared_statements = self.class.type_cast_config_to_boolean(
+            @config.fetch(:prepared_statements) { default_prepared_statements }
+          )
+        end
+
+        def connect!
+          verify!
+          self
+        end
+
+        def reconnect!
+          @lock.synchronize do
+            disconnect!
+            connect
+          rescue StandardError => original_exception
+            @verified = false
+            raise translate_exception_class(original_exception, nil, nil)
+          end
+        end
+
+        def exec_rollback_db_transaction
+          # 16384 tests the bit flag for SERVER_SESSION_STATE_CHANGED, which gets set when the
+          # last statement executed has caused a change in the server's state.
+          if active? || (@raw_connection.server_status & 16_384).positive?
+            super
+          else
+            @verified = false
+          end
+        end
+
+        def with_raw_connection(uses_transaction: true, **_kwargs)
+          @lock.synchronize do
+            @raw_connection = @connection || nil unless instance_variable_defined?(:@raw_connection)
+            unless @verified
+              verify!
+              @verified = true
+            end
```

Comment:
This feels like a partial backport of deferred connection / verification, and possibly some reconnection behaviour. I'm not so keen on that: it used to live in our adapter, and the whole reason I extracted it and pushed it upstream first is to keep that--Trilogy-independent--state management concern out of this gem.

Reply:
> If we remove `@verified` then it would change quite a bit of the fundamental nature of this driver, and require that a lot of test cases be skipped

That's potentially fine... it's very plausible we have remnant test cases from when the GH-internal precursor to this adapter did implement reconnection logic itself, and that they should now go away because they're just testing upstream behaviour.

Reply:
Sounds good -- ended up only needing to skip 3 examples:
* `#execute answers results for valid query after losing connection unexpectedly`
* `#execute fails if the connection is closed`
* `can reconnect after failing to commit`

Perhaps at a minimum these are the specific ones that can now be removed altogether.